src/tactics/FStar.Tactics.CtrlRewrite.fst

```diff
@@ -118,17 +123,18 @@ let __do_rewrite
        BU.print2 "rewrite_rec: succeeded rewriting\n\t%s to\n\t%s\n"
                    (Print.term_to_string tm)
                    (Print.term_to_string ut)) (fun () ->
-    ret ut)))))
+    ret ut))))))
   end
 
 (* If __do_rewrite fails with "SKIP" we do nothing *)
 let do_rewrite
     (g0:goal)
+    (allow_subtyping : bool)
     (rewriter : rewriter_ty)
     (env : env)
     (tm : term)
   : tac term
-= bind (catch (__do_rewrite g0 rewriter env tm)) (function
+= bind (catch (__do_rewrite g0 allow_subtyping rewriter env tm)) (function
        | Inl (TacticFailure "SKIP") -> ret tm
```

Comment:
@R1kM : I think our last conversation about this PR was to try to find a way to avoid the use of matching on this string constant, right?

Reply:
@nikswamy My understanding was that, we would ideally like to avoid relying on this string, but that it might be a bit hard without refactoring ctrl_rewrite, and hence should be done in a subsequent PR since this mechanism was already part of ctrl_rewrite. Some possible ideas to do this I had:
* Raise an actual failure (instead of fail "SKIP") if none of the `l_to_r` lemma rewritings worked, and catch it when calling `t_pointwise` in the implementation of `l_to_r`. The problem is that this prevents deep rewritings, as `l_to_r` will now stop at the first subterm where rewriting is not possible.
* Rollback by default during `ctrl_rewrite` if no rewriting succeeded. I'm not sure this can be distinguished from a standard exception, and there are likely cases where we would want exceptions raised during ctrl_rewrite to bubble up.
* Pattern-match on a new exception (say, `CtrlRewriteSkip`) instead of `TacticFailure "SKIP"`, and raise this exception at the end of `l_to_r` if no rewriting was performed. This might be the better solution, but 1. I'm not sure how removing the pattern-matching on `TacticFailure "SKIP"` would impact existing code, and 2. is it possible to expose not just functions, but exceptions from FStar.Tactics.Basic.fsti, so that they can be accessed in `FStar.Tactics.Builtins`?